pkg/api/validate/dynamic/sku.go

```diff
@@ -0,0 +1,71 @@
+package dynamic
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/computeskus"
+)
+
+// ValidateWorkerSku uses resourceSkusClient to ensure that the VM sizes listed in the cluster document are available for use in the target region.
+func (dv *dynamic) ValidateVmSku(ctx context.Context, location string, subscriptionID string, oc *api.OpenShiftCluster) error {
+	// Get a list of available worker SKUs, filtering by location.
+	filter := fmt.Sprintf("location eq %s", location)
+	skus, err := dv.resourceSkusClient.List(ctx, filter)
+	if err != nil {
+		return err
+	}
+
+	filteredSkus := computeskus.FilterVmSizes(skus, location)
+
+	// Ensure sku in MasterProfile exists in target region
+	masterSkuAvailable := false
```

Comment:
I don't think we need this bool `masterSkuAvailable` since we are returning early on line 32 the check on it on line 43 should always be true.

Reply:
Good catch!

Same is true for `workerSkuAvailable` I believe.